modules/programs/broot.nix

```diff
@@ -0,0 +1,256 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.broot;
+
+  configFile = config:
+    pkgs.runCommand "conf.toml"
+      {
+         buildInputs = [ pkgs.remarshal ];
+         preferLocalBuild = true;
+         allowSubstitutes = false;
+      }
+      ''
+        remarshal -if json -of toml \
+          < ${pkgs.writeText "verbs.json" (builtins.toJSON config)} \
+          > $out
+      '';
+
+  brootConf = {
+    verbs = cfg.verbs;
+    skin = cfg.skin;
+  };
+
+in
+
+{
+  meta.maintainers = [ maintainers.aheaume ];
+
+  options.programs.broot = {
+    enable = mkEnableOption ''
+      A better way to navigate directories.
+    '';
+
+    enableBashIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Bash integration.
+      '';
+    };
+
+    enableZshIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Zsh integration.
+      '';
+    };
+
+    enableFishIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Fish integration.
+      '';
+    };
+
+    verbs = mkOption {
+      type = types.listOf types.attrs;
+      default = [
+        {
+          invocation = "p";
+          execution = ":parent";
+        }
+        {
+          invocation = "edit";
+          shortcut = "e";
+          execution = "$EDITOR {file}";
+        }
+        {
+          invocation = "create {subpath}";
+          execution = "$EDITOR {directory}/{subpath}";
+        }
+        {
+          invocation = "view";
+          execution = "less {file}";
+        }
+      ];
+      example = [
+        {
+          invocation = "p";
+          execution = ":parent";
+        }
+        {
+          invocation = "edit";
+          shortcut = "e";
+          execution = "$EDITOR {file}";
+        }
+        {
+          invocation = "create {subpath}";
+          execution = "$EDITOR {directory}/{subpath}";
+        }
+        {
+          invocation = "view";
+          execution = "less {file}";
+        }
+      ];
+      description = ''
```

Comment:
Descriptions must be formatted in docbook markup. So for this one you'd want something along the lines of
```xml
Define new verbs.
</para><para>
The possible attributes are:
<variablelist>
  <varlistentry>
    <term><literal>invocation</literal> (mandatory)</term>
    <listitem><para>how the verb is called by the user, with placeholders for arguments</para></listitem>
  </varlistentry>
  <varlistentry>
    <term><literal>execution</literal> (mandatory)</term>
    <listitem><para>how the verb is executed</para></listitem>
  </varlistentry>
  <varlistentry>
    <term><literal>shortcut</literal> (optional)</term>
    <listitem><para>an alternate way to call the verb (without the arguments part)</para></listitem>
  </varlistentry>
  <varlistentry>
    <term><literal>leave_broot</literal> (optional)</term>
    <listitem><para>whether to quit broot on execution (default: <literal>true</literal>)</para></listitem>
  </varlistentry>
  <varlistentry>
    <term><literal>from_shell</literal> (optional)</term>
    <listitem><para>whether the verb must be executed from the parent shell (needs br, default: <literal>false</literal>)</para></listitem>
  </varlistentry>
</variablelist>
```